static/app/components/events/attachmentViewers/jsonViewer.tsx

```diff
@@ -1,53 +1,56 @@
 import styled from '@emotion/styled';
 
 import ContextData from 'sentry/components/contextData';
-import DeprecatedAsyncComponent from 'sentry/components/deprecatedAsyncComponent';
 import PreviewPanelItem from 'sentry/components/events/attachmentViewers/previewPanelItem';
 import {
   getAttachmentUrl,
   ViewerProps,
 } from 'sentry/components/events/attachmentViewers/utils';
+import LoadingError from 'sentry/components/loadingError';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
+import {t} from 'sentry/locale';
+import {useApiQuery} from 'sentry/utils/queryClient';
+
+export default function NewJsonViewer(props: ViewerProps) {
+  const query = useApiQuery([getAttachmentUrl(props)], {
+    staleTime: Infinity,
+    retry: false,
+  });
+
+  if (query.isLoading) {
+    return <LoadingIndicator mini />;
+  }
 
-type Props = ViewerProps & DeprecatedAsyncComponent['props'];
-
-type State = DeprecatedAsyncComponent['state'];
-
-export default class JsonViewer extends DeprecatedAsyncComponent<Props, State> {
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    return [
-      [
-        'attachmentJson',
-        getAttachmentUrl(this.props),
-        {headers: {Accept: '*/*; charset=utf-8'}},
```

Comment:
@scttcper looks like this seems to work without the `Accept` header - just want to point that out if you think we should still be adding it. Probably works because the response here is json?

Reply:
i think we want to keep the headers if possible, i think some are octet streams or something

Reply:
Done!